agent/agent.go

```diff
@@ -101,26 +123,51 @@ func DefaultConfig() (Config, error) {
 	}, nil
 }
 
-// Start the Agent which includes the registration against Consul Agent
-func (a *Agent) Start() error {
-	log.Println("Registering the agent service with Consul...")
-	err := a.registerConsulService()
+func DefaultAgentConfig() (Config, error) {
+	hostname, err := os.Hostname()
 	if err != nil {
-		return errors.Wrap(err, "could not register consul service")
+		return Config{}, errors.Wrap(err, "could not read the hostname")
 	}
-	log.Println("Consul service registered.")
 
-	defer func() {
-		log.Println("De-registering the agent service with Consul...")
-		err := a.consul.Agent().ServiceDeregister(a.cfg.InstanceName)
+	return Config{
+		InstanceName: hostname,
+		DiscoveryTTL: 15 * time.Second,
+		CheckerTTL:   10 * time.Second,
+	}, nil
+}
+
+// Start the Agent which includes the registration against Consul Agent
+func (a *Agent) Start() error {
+	status := make(chan int, 1)
+	var wg sync.WaitGroup
+
+	// Attempt to register against the consul client
+	wg.Add(1)
+	go func(wg *sync.WaitGroup) {
+		log.Println("Registering the agent service with Consul...")
+		defer wg.Done()
+		err := a.startRegisterLoop()
 		if err != nil {
-			log.Println("An error occurred while trying to deregisterConsulService the agent service with Consul:", err)
+			log.Fatal("Error while registering consul")
```

Comment:
hmm, I think we should just change `Println` with `Fatal` and leave the error message as it was, otherwise we're not reporting the actual error.

In the future, we might need to avoid `Fatal` and properly handle the error in orchestration with the other goroutines, but it's okay for now.